backend/ddang/src/main/java/com/ddang/ddang/chat/application/dto/CreateChatRoomDto.java

```diff
@@ -0,0 +1,26 @@
+package com.ddang.ddang.chat.application.dto;
+
+import com.ddang.ddang.auction.domain.Auction;
+import com.ddang.ddang.auction.domain.exception.WinnerNotFoundException;
+import com.ddang.ddang.chat.domain.ChatRoom;
+import com.ddang.ddang.chat.presentation.dto.request.CreateChatRoomRequest;
+import com.ddang.ddang.user.domain.User;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+public record CreateChatRoomDto(Long auctionId) {
+
+    public static CreateChatRoomDto from(final CreateChatRoomRequest chatRoomRequest) {
+        return new CreateChatRoomDto(chatRoomRequest.auctionId());
+    }
+
+    public ChatRoom toEntity(final Auction auction) {
+        final Optional<User> nullableWinner = auction.findWinner(LocalDateTime.now());
+        if (nullableWinner.isEmpty()) {
+            throw new WinnerNotFoundException("낙찰자가 존재하지 않습니다");
+        }
+
+        return new ChatRoom(auction, nullableWinner.get());
```

Comment:
```suggestion
        final User winner = auction.findWinner(LocalDateTime.now())
                                           .orElseThrow(() -> new WinnerNotFoundException("낙찰자가 존재하지 않습니다"));

        return new ChatRoom(auction, winner);
```

Optional이 null 체크를 수월하게 하기 위한거라서 위처럼 코드를 작성하신 것은 

```
final User nullableWinner = auction.findWinner(LocalDateTime.now()); // null 반환 가능

if (nullableWinner == null) {
    throw new WinnerNotFoundException("낙찰자가 존재하지 않습니다");
}

return new ChatRoom(auction, nullableWinner);
```

위 코드처럼 그냥 null을 반환하는 것보다 비효율적인 코드인 것 같습니다